### PR TITLE
Typescript: ladders-matchmaker.js

### PR DESCRIPTION
**Pull request by Morfent**

### CODE REVIEW

**Comment by Slayer95** [comment]
All these renames........

``format`` should be used for an actual ``Format`` instance. ``formatid`` is better if it's the well, ID.

File: ladders-matchmaker.js [hunk 1]
```diff
@@ -75,14 +103,24 @@ class Matchmaker {
...
+	/**
+	 * Verifies whether or not a match made between two users is valid.
+	 * @param {Search} search1
+	 * @param {Search} search2
+	 * @param {?User} [user1 = null]
+	 * @param {?User} [user2 = null]
+	 * @param {string} format
+	 * @return {number | false | void}
+	 */
+	matchmakingOK(search1, search2, user1 = null, user2 = null, format) {
```
Comment (Slayer95) [inline ladders-matchmaker.js]: Isn't there an alternative other than using defaults?
``user1`` and ``user2`` being falsey is exceptional (by which I mean *buggy* path). It's odd to make the docs more *extensive* for this case, as if it were supported.

File: ladders-matchmaker.js [hunk 2]
```diff
@@ -75,14 +103,24 @@ class Matchmaker {
...
 		if (!user1 || !user2) {
```
Comment (Slayer95) [inline ladders-matchmaker.js]: I suppose the ``ts-ignore`` thing would be useful here?

**Comment by Zarel** [comment]
Yeah, I don't understand why `formatid` was renamed `format`

File: ladders-matchmaker.js (hunk 1 above)
Comment (Morfent) [reply ladders-matchmaker.js]: I suppose it's unnecessary, since if either of the two are undefined the crashlogger would complain about it

**Pull request merged**